Dockerfile.envoy

```diff
@@ -1,9 +1,6 @@
 FROM debian:sid as builder
```

Comment:
we should drop this and only have `Dockerfile.openssl` and `Dockerfile.boringssl`  that are clearlinux based.

offtopic: we should add Github actions build matrix for `openssl` and `boringssl`.

Reply:
> we should drop this and only have `Dockerfile.openssl` and `Dockerfile.boringssl` that are clearlinux based.

I'd rather do it in a followup PR.

Reply:
yes makes sense